Remotion/Data/DomainObjects.UnitTests/Tracing/TracingDataReaderTest.cs

```diff
@@ -603,5 +644,231 @@ public void ReadAndClose ()
       sequence.Verify();
     }
 
+    [Test]
+    public void ReadAsync_HasRecord ()
+    {
+      var token = CancellationToken.None;
+      _innerDataReader.Setup(mock => mock.ReadAsync(token)).Returns(Task.FromResult(true)).Verifiable();
+
+      var hasRecord = _dataReader.ReadAsync(token);
+
+      _innerDataReader.Verify();
+      _extensionMock.Verify();
+      Assert.That(hasRecord.Result, Is.True);
+    }
+
+    [Test]
+    public void ReadAsync_NoRecord ()
+    {
+      var token = CancellationToken.None;
+      _innerDataReader.Setup(mock => mock.ReadAsync(token)).Returns(Task.FromResult(false)).Verifiable();
+
+      var hasRecord = _dataReader.ReadAsync(token);
+
+      _innerDataReader.Verify();
+      _extensionMock.Verify();
+      Assert.That(hasRecord.Result, Is.False);
+    }
+
+    [Test]
+    public async Task ReadAsyncAndCloseAsync ()
+    {
+      var token = CancellationToken.None;
+      var sequence = new VerifiableSequence();
+      _innerDataReader.InVerifiableSequence(sequence).Setup(mock => mock.ReadAsync(token)).Returns(Task.FromResult(true)).Verifiable();
+      _extensionMock
+          .InVerifiableSequence(sequence)
+          .Setup(mock => mock.QueryCompleted(_connectionID, _queryID, It.Is<TimeSpan>(p => p > TimeSpan.Zero), 1))
+          .Verifiable();
+      _innerDataReader.InVerifiableSequence(sequence).Setup(mock => mock.CloseAsync()).Returns(Task.CompletedTask).Verifiable();
+
+      await _dataReader.ReadAsync(token);
+      await _dataReader.CloseAsync();
+
+      _innerDataReader.Verify();
+      _extensionMock.Verify();
+      sequence.Verify();
+    }
+
+    [Test]
+    public void VisibleFieldCount ()
+    {
+      var assumedResult = 73;
+
+      _innerDataReader.Setup(mock => mock.VisibleFieldCount).Returns(assumedResult).Verifiable();
+
+      var result = _dataReader.VisibleFieldCount;
+
+      _innerDataReader.Verify();
+      _extensionMock.Verify();
+      Assert.That(result, Is.EqualTo(assumedResult));
+    }
+
+    [Test]
+    public void NextResultAsync ()
```

Comment:
member ordering. please check that the async and sync methods are groups. NextResultAsync is not.